synapse/lib/agenda.py

```diff
@@ -863,19 +870,24 @@ async def _runJob(self, user, appt):
                     # we only care about the last five errors
                     'lasterrs': list(appt.lasterrs[-5:]),
                 }
-                await self.core.addCronEdits(appt.iden, edits)
+
+                if self.core.isactive:
```

Comment:
You might as well move this up to be part of L866 since everything in this block supports the edits.

```
    if not success and self.core.isactive:
```

Reply:
Actually, this block could probably be moved down and combined with the logic between the log messages and the isfini check.

Reply:
After talking about this, we decided to leave this as-is since we've got a backlog item for refactoring the Cron system and would involve unwinding the double nexus entries there for us.